Adjust for Django 3.0 deprecations - part 1

millerdev opened the pull request:
- The `smart_text()` and `force_text()` aliases (since Django 2.0) of `smart_str()` and `force_str()` are deprecated.
- `django.utils.http.urlquote()`, `urlquote_plus()`, `urlunquote()`, and `urlunquote_plus()` are deprecated in favor of the functions that they’re aliases for: [`urllib.parse.quote()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote), [`quote_plus()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.quote_plus), [`unquote()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote), and [`unquote_plus()`](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.unquote_plus).
- `django.utils.translation.ugettext()`, `ugettext_lazy()`, `ugettext_noop()`, `ungettext()`, and `ungettext_lazy()` are deprecated in favor of the functions that they’re aliases for: [`django.utils.translation.gettext()`](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.translation.gettext), [`gettext_lazy()`](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.translation.gettext_lazy), [`gettext_noop()`](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.translation.gettext_noop), [`ngettext()`](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.translation.ngettext), and [`ngettext_lazy()`](https://docs.djangoproject.com/en/4.0/ref/utils/#django.utils.translation.ngettext_lazy).

:blowfish: Review by commit.

## Safety Assurance

### Safety story

Lots of files being changed here, but the changes are mostly mechanical. All `ugettext.*` replacements were done automated find/replace on whole words. Lint cleanups were done with flake8 with a few minor by-hand adjustments for readability.

### Automated test coverage

No new tests.

### QA Plan

Does this need QA? I'm leaning toward no.

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

millerdev committed:
(smart|force)_text -> (smart|force)_str

millerdev committed:
Lint

millerdev committed:
django.utils.http Django 3.0 deprecations

django.utils.http.urlquote(), urlquote_plus(), urlunquote(), and urlunquote_plus() are deprecated in favor of the functions that they’re aliases for: urllib.parse.quote(), quote_plus(), unquote(), and unquote_plus()

https://docs.djangoproject.com/en/4.0/releases/3.0/#id3

millerdev committed:
ugettext_lazy -> gettext_lazy

django.utils.translation.ugettext(), ugettext_lazy(), ugettext_noop(), ungettext(), and ungettext_lazy() are deprecated in favor of the functions that they’re aliases for: django.utils.translation.gettext(), gettext_lazy(), gettext_noop(), ngettext(), and ngettext_lazy().

https://docs.djangoproject.com/en/4.0/releases/3.0/#id3

millerdev committed:
Lint

millerdev committed:
ugettext_noop -> gettext_noop

django.utils.translation.ugettext(), ugettext_lazy(), ugettext_noop(), ungettext(), and ungettext_lazy() are deprecated in favor of the functions that they’re aliases for: django.utils.translation.gettext(), gettext_lazy(), gettext_noop(), ngettext(), and ngettext_lazy().

https://docs.djangoproject.com/en/4.0/releases/3.0/#id3

millerdev committed:
Lint

millerdev committed:
ugettext -> gettext

django.utils.translation.ugettext(), ugettext_lazy(), ugettext_noop(), ungettext(), and ungettext_lazy() are deprecated in favor of the functions that they’re aliases for: django.utils.translation.gettext(), gettext_lazy(), gettext_noop(), ngettext(), and ngettext_lazy().

https://docs.djangoproject.com/en/4.0/releases/3.0/#id3

millerdev committed:
Lint

millerdev committed:
ungettext -> ngettext

django.utils.translation.ugettext(), ugettext_lazy(), ugettext_noop(), ungettext(), and ungettext_lazy() are deprecated in favor of the functions that they’re aliases for: django.utils.translation.gettext(), gettext_lazy(), gettext_noop(), ngettext(), and ngettext_lazy().

https://docs.djangoproject.com/en/4.0/releases/3.0/#id3

millerdev committed:
Fix bad import

mkangia commented on corehq/apps/accounting/forms.py:
```diff
@@ -343,8 +343,8 @@ def clean_enterprise_restricted_signup_domains(self):
-                raise ValidationError("The following domains are already restricted by another account: " +
-                                      ", ".join(errors))
+                raise ValidationError("The following domains are already restricted by another account: "
+                                      + ", ".join(errors))
```
do we need a "+" here?

mkangia reviewed (approved):
Looks good.

There were already a lot of changes here due to deprecations so not sure if including lint fixes was really ideal.

For sanity, if we don't want to do a QA, Do we want to keep this on staging for a while before merging?

millerdev commented:
> not sure if including lint fixes was really ideal.

I vacillated on this, but decided lint fixes were better than tons of stickler warnings on the PR. That definitely means it's best to review commits individually. :tropical_fish: 

> Do we want to keep this on staging for a while before merging?

Yes, I will put this on staging.

millerdev committed:
Merge branch 'master' into dm/django3.0-part1

millerdev committed:
Merge remote-tracking branch 'origin/master' into dm/django3.0-part1

millerdev committed:
ugettext -> gettext

Introduced by commits 7fd573ebac47b0bad82455cf23de076911083b0c and 660fbfb64de6b917fbb895d8feffe4d4e50f8c07

gherceg committed:
Merge branch 'master' into dm/django3.0-part1

millerdev committed:
Merge branch 'master' into dm/django3.0-part1

millerdev committed:
Merge remote-tracking branch 'origin/master' into dm/django3.0-part1

millerdev committed:
Merge branch 'master' into dm/django3.0-part1
